src/main/java/ua/goit/gojava32/kickstarter/controller/ProjectController.java

```diff
@@ -0,0 +1,111 @@
+package ua.goit.gojava32.kickstarter.controller;
+
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.ModelAndView;
+
+import ua.goit.gojava32.kickstarter.model.Category;
+import ua.goit.gojava32.kickstarter.model.Project;
+import ua.goit.gojava32.kickstarter.service.CategoryService;
+import ua.goit.gojava32.kickstarter.service.CommentService;
+import ua.goit.gojava32.kickstarter.service.ProjectService;
+import ua.goit.gojava32.kickstarter.service.UserService;
+
+import java.security.Principal;
+import java.util.List;
+
+@Controller
+public class ProjectController {
+
+  @Autowired
+  ProjectService projectService;
+  @Autowired
+  CommentService commentService;
+  @Autowired
+  CategoryService categoryService;
+  @Autowired
+  UserService userService;
+
+  @RequestMapping(value = "/project/{id}")
+  @ResponseBody
+  public ModelAndView showProject(@PathVariable("id") int id, Principal principal){
+    Project project = projectService.get(id);
+    ModelAndView vm = new ModelAndView("project");
+
+    vm.addObject("project", project);
+    vm.addObject("category", project.getCategory());
+    vm.addObject("comments", projectService.getProjectComments(project));
+    vm.addObject("blogs", projectService.getProjectBlogPosts(project));
+    if (principal != null && (project.getUser().getEmail()).equals(principal.getName())) {
+      vm.addObject("showAddBlog", true);
+    } else {
+      vm.addObject("showAddBlog", false);
+    }
+
+    return vm;
+  }
+
+  @RequestMapping(value = "/project/{id}/delete")
+  @ResponseBody
+  public ModelAndView deleteProject(@PathVariable("id") int id, Principal principal){
+    Project project = projectService.get(id);
+    ModelAndView vm;
+    if (principal != null && (project.getUser().getEmail()).equals(principal.getName())) {
+      project = projectService.delete(project);
+      vm = new ModelAndView("redirect:/profile");
+    } else {
+      vm = new ModelAndView("redirect:/error");
```

Comment:
/error is duplicated in a lot of places and looks like a candidate to become a constant